### ISSUE TITLE
Download

### ISSUE THREAD

**Issue by MadVanz** [BeamlakAschalew/cinemax#37]
### Steps to reproduce

Download doesn't work

### Expected behavior

It should download

### Actual behavior

It didn't download

### Android version

Android 11

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by BeamlakAschalew** [comment]
The download button on the current player doesn't work because I have disabled any navigation towards outside to block pop-ups, even if I enable navigation, the download button doesn't really work. But since I found a direct video source users will have a native player instead of webview in the future. And if I found a Flutter library to convert m3u8 files into mp4, mkv or whatever, It'll have a properly working download option with a dedicated download manager etc.

**Comment by DarkCrypt** [comment]
> And if I found a Flutter library to convert m3u8 files into mp4, mkv or whatever, It'll have a properly working download option with a dedicated download manager etc.

For converting and downloading videos you could use [FFmpeg-Kit](https://github.com/arthenica/ffmpeg-kit) for Flutter. Sounds like it could be super bloated and slow, idk. Maybe there's alternatives?

**Comment by BeamlakAschalew** [comment]
> > And if I found a Flutter library to convert m3u8 files into mp4, mkv or whatever, It'll have a properly working download option with a dedicated download manager etc.
> 
> For converting and downloading videos you could use [FFmpeg-Kit](https://github.com/arthenica/ffmpeg-kit) for Flutter. Sounds like it could be super bloated and slow, idk. Maybe there's alternatives?

Yeah I also found this but it needs modifications (VLC uses FFmpeg but it chocks when you convert m3u8 to mp4) otherwise it's gonna be bloated and broken as you said. It also takes time but I'll announce progresses made.
